### PR TITLE
Improve layout responsiveness on small screens

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Some conversations about timeline earlier this week have made it clear that we are going to have to shift to a mobile-first development approach in order to hit our release target. To that end I have done a pass over what's in `master` so far to make sure the small-screen layout is at least functional (if not 100% polished, pending the next round of mocks). 

This mainly involves:
- making sure side-by-side elements stack vertically when necessary (mainly through creative use of flexbox)
- making sure charts properly grow and shrink (some are more amenable to this than others; I have been using [react-measure](https://github.com/souporserious/react-measure) for the ones that require an explicit dynamic width)

Going forward, once this is merged, we should have a reasonable enough base to stop punting on responsive design issues when adding new charts. (The V1 strategy, briefly, is to adjust the layout or properties of charts as they shrink without actually replacing them with entirely different charts. As a last resort, if we find them entirely unworkable, we may hide them below a certain size and replace them with simple number readouts or something like that.)

A few things are still notably broken at this point; they may change before this PR comes out of draft mode or later, depending on how things go:
- the navigation menu fits on the screen but looks bad and has weird spacing; its' probably going to become collapsible
- ~the bubble chart fits on the screen but isn't readable; this has already been discussed with Juan, we will have to employ a different circle-packing algorithm on small screens~
- ~changing the demographic dimension in the revocations section causes the screen to jump back up to the top; this only happens on mobile, not sure what the story is there~ (I think this is just some kind of bug in the Firefox dev tools, I can't reproduce it in any other browser)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

none

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by macfarlandian** [comment]
this PR still doesn't address the issue with the proportional bar height; other than that and the menu issue noted in the description, this should have the latest state of the parole page looking acceptable on screens as narrow as 360px

File: src/viz-parole-population/VizParolePopulation.js [hunk 1]
```diff
@@ -28,10 +29,23 @@ const VizParolePopulationContainer = styled.div`
...
+  display: flex;
+  flex: 1 1 auto;
+  margin-bottom: 24px;
+  min-width: 340px;
+  width: calc((100% - ${GUTTER_WIDTH}) / 2);
+`;
+
+const Gutter = styled.div`
+  height: 1px;
+  width: ${GUTTER_WIDTH};
```
Comment (sprad) [inline src/viz-parole-population/VizParolePopulation.js:48]: Padding be damned

File: src/viz-parole-population/VizParolePopulation.js [hunk 2]
```diff
@@ -156,15 +169,26 @@ export default function VizParolePopulation({
...
-        </MapWrapper>
+        <Measure bounds>
+          {({
+            measureRef,
+            contentRect: {
+              bounds: { width },
+            },
+          }) => (
+            <MapWrapper ref={measureRef}>
+              <StateDistrictMap
```
Comment (sprad) [inline src/viz-parole-population/VizParolePopulation.js:180]: The map seems to have stopped filling up the `VizWrapper` on larger screens:

Screen Shot 2020-07-15 at 7 50 27 PM

Comment (macfarlandian) [reply src/viz-parole-population/VizParolePopulation.js:180]: damn you are right ... it flexes down but it never flexes up again 😡

File: src/viz-parole-population/VizParolePopulation.js (hunk 1 above)
Comment (macfarlandian) [reply src/viz-parole-population/VizParolePopulation.js:48]: this is not my proudest achievement ...

File: src/bubble-chart/BubbleChart.js [hunk 3]
```diff
@@ -1,15 +1,26 @@
 import PropTypes from "prop-types";
 import { cumsum } from "d3-array";
-import { forceSimulation, forceX, forceY } from "d3-force";
+import { forceCollide, forceSimulation, forceX, forceY } from "d3-force";
```
Comment (sprad) [inline src/bubble-chart/BubbleChart.js:3]: Dude, these Bubble animations are SO good

bubbles

File: src/bubble-chart/BubbleChart.js [hunk 4]
```diff
@@ -22,57 +33,96 @@ const BubbleValueLabel = styled.text`
...
+      .force(
+        "limit",
+        forceLimit()
+          .radius(getRadius)
+          .x0(margin.left)
+          .x1(width - margin.right)
+          .y0(margin.top)
+          .y1(height - margin.bottom)
+      )
+      // this force prevents them from overlapping
```
Comment (sprad) [inline src/bubble-chart/BubbleChart.js:123]: These comments are very helpful

**Pull request merged**